cache.go

```diff
@@ -15,6 +15,7 @@ const (
 	EvictionReasonDeleted EvictionReason = iota + 1
 	EvictionReasonCapacityReached
 	EvictionReasonExpired
+	EvictionReasonTotalCostExceeded
```

Comment:
```suggestion
	EvictionReasonMaxCostExceeded
```

To keep things consistent with the options.

Reply:
done